lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2522,6 +2522,18 @@ namespace {
           VarDecl *pointeeProperty =
               synthesizer.makeDereferencedPointeeProperty(
                   getterAndSetter.first, getterAndSetter.second);
+
+          // Import the attributes from dereference operator clang decl to
+          // pointee property swift decl.
+          FuncDecl *setterImpl = getterAndSetter.second;
+          FuncDecl *getterImpl = getterAndSetter.first ? getterAndSetter.first
+                                                       : getterAndSetter.second;
+          if (auto clangDeclForOperatorStar =
+                  dyn_cast_or_null<clang::NamedDecl>(
+                      getterImpl->getClangDecl())) {
+            Impl.importAttributes(clangDeclForOperatorStar, pointeeProperty);
+          }
+
           result->addMember(pointeeProperty);
         }
       }
```

Comment:
This logic looks quite specific to the synthesized `pointee` property. Are there other synthesized properties where this is necessary? If so, I wonder if we can factor this out in to a method and call this from other places as well.

Reply:
I discussed this with @egorzhdan and @beccadax . I think this problem should be there for other synthesized properties also (like computed properties). But I wanted to get feedback from people by making a patch only for pointee property and then think about solving this problem in general for all synthesized properties.

Reply:
@ravikandhadai I factored out the logic in a helper function and called it fro pointee and successor for now. I'll fix for other cases of synthesized properties in a later patch.